main.go

```diff
@@ -97,20 +124,30 @@ func main() {
 		}
 	}
 
-	ctx := context.TODO()
-	// Become the leader before proceeding
-	err = leader.Become(ctx, "policy-status-sync-lock")
+	hubClient, err := client.New(hubCfg, client.Options{Scheme: scheme})
 	if err != nil {
-		log.Error(err, "")
+		log.Error(err, "Failed to generate client to the hub cluster")
+		os.Exit(1)
+	}
+	var kubeClient kubernetes.Interface = kubernetes.NewForConfigOrDie(hubCfg)
+
+	eventBroadcaster := record.NewBroadcaster()
+	namespace, err := GetWatchNamespace()
+	if err != nil {
+		log.Error(err, "Failed to get watch namespace")
 		os.Exit(1)
 	}
+	eventBroadcaster.StartRecordingToSink(&corev1.EventSinkImpl{Interface: kubeClient.CoreV1().Events(namespace)})
+	hubRecorder := eventBroadcaster.NewRecorder(eventsScheme, v1.EventSource{Component: sync.ControllerName})
 
-	// Set default manager options
 	options := manager.Options{
-		Namespace:          namespace,
+		LeaderElection:     tool.Options.EnableLeaderElection,
+		LeaderElectionID:   "policy-status-sync.open-cluster-management.io",
 		MetricsBindAddress: "0",
+		Namespace:          namespace,
+		Port:               9443,
+		Scheme:             scheme,
 	}
```

Comment:
Comparing this to the propagator, I notice there is no `HealthProbeBindAddress` - it might be that the propagator doesn't actually use that health probe anyway (I don't see it in the deployment yaml) but I'm still curious why it's missing here. Is it deliberate?

It also seems odd to specify "0" for the `MetricsBindAddress` - does that turn off the out-of-the-box metrics generated by controller-runtime (or whatever library)?

Finally, `Port`. What does that do here? It is not present in the propagator.

Reply:
+1 for using a useful name for the leader election ID, I should've done that in propagator.